### PR TITLE
Container recreate stateless test

### LINKED ISSUE #300: Do not return 410 for "The requested resource is no longer available at the server." error
>We are continuously addressing and improving the SDK, if possible, make sure the problem persist in the [latest SDK version](https://www.nuget.org/packages/Microsoft.Azure.Cosmos).

**Describe the bug**
When a collection is dropped and recreated with the same name, certain codepaths return 410 status code back to the caller, which is not expected

Error from the store: Status Code=410, ActivityID=69099556-7c56-4b05-9a54-a420eb53e30c, Additional Details='Message: {"Errors":["The requested resource is no longer available at the server."]}ActivityId: 69099556-7c56-4b05-9a54-a420eb53e30c, Request URI: /apps/a4486783-0b00-4778-b422-99ce68be0b8b/services/a535723c-b5d1-43bc-a60d-db9697bd2047/partitions/538fc5fc-8c97-4d2f-9c74-b04fb10cf515/replicas/132014017409262082s/, RequestStats: RequestStartTime: 2019-05-06T20:31:13.7199620Z, RequestEndTime: 2019-05-06T20:31:13.7355740Z, Number of regions attempted: 1ResponseTime: 2019-05-06T20:31:13.7355740Z, StoreResult: StorePhysicalAddress: rntbd://cdb-ms-prod-westus1-fd22.documents.azure.com:14163/apps/a4486783-0b00-4778-b422-99ce68be0b8b/services/a535723c-b5d1-43bc-a60d-db9697bd2047/partitions/538fc5fc-8c97-4d2f-9c74-b04fb10cf515/replicas/132014017349730354p/, LSN: 6, GlobalCommittedLsn: 6, PartitionKeyRangeId: , IsValid: True, StatusCode: 410, IsGone: True, IsNotFound: False, IsInvalidPartition: True, RequestCharge: 0, ItemLSN: -1, SessionToken: , ResourceType: Document, OperationType: ReadFeedResponseTime: 2019-05-06T20:31:13.7355740Z, StoreResult: StorePhysicalAddress: rntbd://cdb-ms-prod-westus1-fd22.documents.azure.com:14022/apps/a4486783-0b00-4778-b422-99ce68be0b8b/services/a535723c-b5d1-43bc-a60d-db9697bd2047/partitions/538fc5fc-8c97-4d2f-9c74-b04fb10cf515/replicas/132014017409262082s/, LSN: 6, GlobalCommittedLsn: 6, PartitionKeyRangeId: , IsValid: True, StatusCode: 410, IsGone: True, IsNotFound: False, IsInvalidPartition: True, RequestCharge: 0, ItemLSN: -1, SessionToken: , ResourceType: Document, OperationType: ReadFeed, SDK: Windows/10.0.14393 cosmos-netstandard-sdk/3.0.0'" Argument1="" TraceSource="CosmosDBTrace"

**To Reproduce**
Have multiple SDK clients querying data from a given collection (so each SDK client has its own cache).  Drop and recreate the collection with the same name using one of the client.  Rest of the clients will randomly fail with these errors in the QUERY/READ path

**Expected behavior**

**Actual behavior**
Provide a description of the actual behavior observed. 

**Environment summary**
SDK Version:
OS Version (e.g. Windows, Linux, MacOSX)

**Additional context**
Add any other context about the problem here (for example, complete stack traces or logs).

### PR DESCRIPTION
**Pull request by kirankumarkolli**
# Pull Request Template

## Description

When a collection is dropped and recreated with the same name, certain codepaths return 410 status code back to the caller, which is not expected.

Query code paths are returning 410 status code.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Closing issues
closes #300 

## Assignee

Please add yourself as the assignee

## Porjects

Please add relevant projects so this issue can be properly tracked.

### CODE REVIEW

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs [hunk 1]
```diff
@@ -1031,6 +1031,78 @@ public async Task MigrateDataInNonPartitionContainer()
...
+            {
+                string dbName = Guid.NewGuid().ToString();
+                string containerName = Guid.NewGuid().ToString();
+
+                CosmosDatabase db1 = await cc1.Databases.CreateDatabaseAsync(dbName);
+                CosmosContainer container1 = await db1.Containers.CreateContainerAsync(containerName, "/id");
+
+                CosmosResponseMessage response = await container1.Items.ReadItemStreamAsync("any", "any");
+                Assert.AreEqual(HttpStatusCode.NotFound, response.StatusCode);
+                await CosmosItemTests.ExecuteQueryAsync(container1, HttpStatusCode.Accepted);
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs]: release branch passes with OK status code. Seems like regression in master.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs [hunk 2]
```diff
@@ -1031,6 +1031,78 @@ public async Task MigrateDataInNonPartitionContainer()
...
+                Assert.AreEqual(HttpStatusCode.NotFound, response.StatusCode);
+                await CosmosItemTests.ExecuteQueryAsync(container2, HttpStatusCode.Accepted);
+
+                // Delete container 
+                await container1.DeleteAsync();
+
+                // Read on deleted container through client1
+                response = await container1.Items.ReadItemStreamAsync("any", "any");
+                Assert.AreEqual(HttpStatusCode.NotFound, response.StatusCode);
+                await CosmosItemTests.ExecuteQueryAsync(container1, HttpStatusCode.NotFound);
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs]: Failing in master but passing in release branch. @j82w  can u please look into it.

Comment (j82w) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs]: That is what this [PR](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/281) is for. I already merged a different PR for release to unblock compute.

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs]: Here is a simplified test-case which repro's GONE exception. 
It also repros in release branch as well. 

Am suspecting that PartitionKeyRangeGoneRetryPolicy is not used in query pipeline.

**Comment by j82w** [comment]
Moved this to a new branch and PR. https://github.com/Azure/azure-cosmos-dotnet-v3/pull/373

**Pull request closed without merging**